skimage/color/colorconv.py

```diff
@@ -1179,7 +1179,10 @@ def _lab2xyz(lab, illuminant, observer):
     n_invalid = invalid[0].size
     if n_invalid != 0:
         # Warning should be emitted by caller
-        z[invalid] = 0
+        if z.ndim > 0:
+            z[invalid] = 0
+        else:
+            z = 0
```

Comment:
Hmm, I saw this to late but if I am honest I would have preferred to introduce a 
```python
if arr.shape == (3,):
    arr = arr.reshape((1, 3))
```
in `_prepare_colorarray` so that `z` is just always an array which can be assigned to like I indicated in https://github.com/scikit-image/scikit-image/issues/7105#issuecomment-1689990548. Though, it's done now and probably not worth the additional work now. ;)

Reply:
Thanks anyway! :)

Reply:
> Hmm, I saw this to late but if I am honest I would have preferred to introduce a
> 
> ```python
> if arr.shape == (3,):
>     arr = arr.reshape((1, 3))
> ```
> 
> in `_prepare_colorarray` so that `z` is just always an array which can be assigned to like I indicated in [#7105 (comment)](https://github.com/scikit-image/scikit-image/issues/7105#issuecomment-1689990548).

Right, this would be more robust.

> Though, it's done now and probably not worth the additional work now. ;)

It *is* worth it! It's never too late to submit another PR with a better fix :)

Thank *you*, @lagru.

Reply:
Created a new PR with the prefered fixed. 
Thanks @lagru

Reply:
You are completely right @decorouz. I saw your comment after I made this one https://github.com/scikit-image/scikit-image/pull/7127#issuecomment-1712171961, but it seems we see the same problem. But still thank you very much for quickly exploring this option!